src/hooks/remove-fields.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Remove provided field or fields from response.
+ * @param {String or Array} fieldsToRemove a field name String or array
+ *                   of field name Strings to remove from the results.
+ *
+ * after
+ * all, find, get, create, update, patch, remove
+ */
+const removeFields = function(fieldsToRemove){
+  if (!fieldsToRemove) {
+    throw new Error('You must pass a string or array of strings to the "removeFields" hook.');
+  }
+
+  if (!Array.isArray(fieldsToRemove)) {
+    fieldsToRemove = [fieldsToRemove];
+  }
+
+  fieldsToRemove.forEach(field => {
+    if (typeof field !== 'string') {
+      throw new Error('"removeFields" hook: field names must be strings.');
+    }
+  });
+
+  return function (hook, next) {
+    // Skip removing fields if it's flagged as internal.
+    if (!hook.params.internal) {
```

Comment:
I'm wondering if we should be more explicit. `hook.params.internal` makes sense to us but probably doesn't make sense to anyone new to Feathers. Possibly it should be called `isPublic`, `isPrivate`? Just spit ballin' here....